src/message.rs

```diff
@@ -11,14 +12,46 @@ use fvm_shared::{
 use multihash_codetable::{Code, MultihashDigest as _};
 use serde::{Deserialize, Serialize};
 
-fn from_cbor_blake2b256<S: serde::ser::Serialize>(obj: &S) -> Result<Cid, Error> {
+fn from_cbor_blake2b256<S: serde::ser::Serialize>(
+    obj: &S,
+) -> Result<Cid, fvm_ipld_encoding::Error> {
     let bytes = fvm_ipld_encoding::to_vec(obj)?;
     Ok(Cid::new_v1(
         fvm_ipld_encoding::DAG_CBOR,
         Code::Blake2b256.digest(&bytes),
     ))
 }
 
+fn check_address_prefix(s: &str, n: Network) -> bool {
+    if s.len() == 0 || s.len() < 2 {
+        return false;
+    }
+
+    match n {
+        Network::Mainnet => s[0..1].eq("f") || s[0..2].eq("0x"),
+        Network::Testnet => s[0..1].eq("t") || s[0..2].eq("0x"),
+    }
+}
+
+pub fn parse_address(s: &str, n: Network) -> anyhow::Result<Address> {
+    if !check_address_prefix(s, n) {
+        return Err(anyhow!("Wrong Network"));
+    }
+
+    return match n.parse_address(s) {
+        Ok(addr) => Ok(addr),
+        Err(_e) => {
+            // Try parsing as 0x ethereum address
+            if s.len() != 42 {
+                return Err(anyhow!("Invalid Address"));
+            }
+
+            let addr = hex::decode(&s[2..])?;
+            Ok(Address::new_delegated(10, &addr)?)
+        }
```

Comment:
`10` and `42` are magic numbers and require quite a specific knowledge to identify at first glance. It'd help the reader to properly name them.